### PR TITLE
fixes crash when editing input fields

### LINKED ISSUE #6: Multiple TextField support
The current `textModel` is only working when there is one TextField displayed.

Reproduce the bug by using the [simpleDemo](https://github.com/Drakirus/go-flutter-desktop-embedder/tree/master/example/simpleDemo) example. The embedder crash when words are typed in a TextField when there is another TextField displayed on the screen that has been filled.  

```flutter: Observatory listening on http://127.0.0.1:37183/
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/Drakirus/go-flutter-desktop-embedder.(*textModel).addChar(0x634080, 0xc0000a9c74, 0x1, 0x1)
        /home/drakirus/lab/go/src/github.com/Drakirus/go-flutter-desktop-embedder/textModel.go:32 +0x15e
github.com/Drakirus/go-flutter-desktop-embedder.glfwCharCallback(0xc0000b6000, 0x66)
        /home/drakirus/lab/go/src/github.com/Drakirus/go-flutter-desktop-embedder/gutter.go:198 +0x76
github.com/go-gl/glfw/v3.2/glfw.goCharCB(0xef9990, 0x66)
        /home/drakirus/lab/go/src/github.com/go-gl/glfw/v3.2/glfw/input.go:294 +0x51
github.com/go-gl/glfw/v3.2/glfw._cgoexpwrap_f7a05a868eb7_goCharCB(0xef9990, 0x66)
        _cgo_gotypes.go:2147 +0x33
github.com/go-gl/glfw/v3.2/glfw._Cfunc_glfwPollEvents()
        _cgo_gotypes.go:1196 +0x41
github.com/go-gl/glfw/v3.2/glfw.PollEvents()
        /home/drakirus/lab/go/src/github.com/go-gl/glfw/v3.2/glfw/window.go:777 +0x20
github.com/Drakirus/go-flutter-desktop-embedder.Run(0xc0000a9f70, 0x3, 0x3, 0x0, 0x0)
        /home/drakirus/lab/go/src/github.com/Drakirus/go-flutter-desktop-embedder/gutter.go:80 +0x21d
main.main()
        /home/drakirus/lab/go/src/go-flutter-desktop-embedder/example/demo/main.go:25 +0xc6
```

### PR DESCRIPTION
**Pull request by james-lawrence**
- prefix text input methods uniformly.
- use a switch statement for selecting method behaviour.
- fix crashes in text input by properly setting initial
position of the cursor.

may fix #6, I didn't notice any issues with multiple text inputs. however I did notice issues when hitting backspace or inserting text. backspace would crash, and text inserts would be in the incorrect position.

### CODE REVIEW

File: flutter/message.go [hunk 1]
```diff
@@ -15,9 +15,9 @@ const (
 	TextUpdateStateMethod = "TextInputClient.updateEditingState"
 
 	// text
-	SetClientMethod       = "TextInput.setClient"
```
Comment (james-lawrence) [inline flutter/message.go]: should add a updateState method to textModel and handle all these methods in there.

File: gutter.go [hunk 2]
```diff
@@ -271,26 +271,24 @@ func onPlatformMessage(platMessage flutter.PlatformMessage, window unsafe.Pointe
...
+		switch message.Method {
+		case flutter.TextInputClientClear:
 			state.clientID = 0
-		}
-
-		if message.Method == flutter.SetClientMethod {
+		case flutter.TextInputClientSet:
 			var body []interface{}
 			json.Unmarshal(message.Args, &body)
 			state.clientID = body[0].(float64)
```
Comment (james-lawrence) [inline gutter.go]: *facepalm* why couldn't they just include the ID in the initial structure? lol

File: gutter.go [hunk 3]
```diff
@@ -271,26 +271,24 @@ func onPlatformMessage(platMessage flutter.PlatformMessage, window unsafe.Pointe
...
-		}
-
-		if message.Method == flutter.SetEditingStateMethod {
+		case flutter.TextInputSetEditState:
 			if state.clientID != 0 {
```
Comment (james-lawrence) [inline gutter.go]: not sure this check is strictly necessary. but left it in anyways.

**Comment by Drakirus** [comment]
It does fix #6 :+1:

**Pull request merged**